src/redux/search/actionCreater.js

```diff
@@ -0,0 +1,41 @@
+import { API_KEY, API_URL } from 'react-native-dotenv';
+
+export const SEARCH_START = 'search_start';
+export const SEARCH_SUCCESS = 'search_success';
+export const SEARCH_FAILED = 'search_failed';
+
+export function searchStart() {
+  return {
+    type: SEARCH_START,
+  };
+}
+
+export function searchSuccess(result) {
+  return {
+    type: SEARCH_SUCCESS,
+    payload: result,
+  };
+}
+
+export function searchFailed(error) {
+  return {
+    type: SEARCH_FAILED,
+    payload: error,
+  };
+}
+
+export function searchData(input) {
+  return (dispatch) => {
+    const url = new URL(`${API_URL}/search/${input}`);
+    url.searchParams.append('token', API_KEY);
```

Comment:
You should dispatch a search start action here
```suggestion
    url.searchParams.append('token', API_KEY);
    dispatch(searchStart());
```